include/dlaf/eigensolver/bt_reduction_to_band.h

```diff
@@ -0,0 +1,114 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <blas.hh>
+
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/eigensolver/bt_reduction_to_band/impl.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace eigensolver {
+
+/// Eigenvalue back-transformation implementation on local memory.
+///
+/// It computes Q C, where Q = HH(1) HH(2) ... HH(m-b)
+/// (HH(j) is the House-Holder transformation (I - v tau vH)
+/// defined by the j-th element of tau and the HH reflector stored in the j-th column of the matrix V.
+///
+/// @param mat_c contains the (m x n) matrix C (blocksize (mb x nb)), while on exit it contains Q C.
+/// @param mat_v is (m x m) matrix with blocksize (mb x mb), which contains the Householder reflectors.
+/// The j-th HH reflector is v_j = (1, V(mb + j : n, j)).
+/// @param taus is a (blocked) vector of size m (blocksize mb). The j-th element is the scaling factor
+/// for the j-th HH tranformation.
+/// @pre mat_c is not distributed,
+/// @pre mat_v is not distributed.
+template <Backend backend, Device device, class T>
+void backTransformationReductionToBand(
+    const SizeType b, Matrix<T, device>& mat_c, Matrix<const T, device>& mat_v,
+    common::internal::vector<pika::shared_future<common::internal::vector<T>>> taus) {
```

Comment:
Is there a plan for that `internal::vector<shared_future>` in the API? Is it going to become non-internal or disappear from the signature at some point?

Reply:
The plan is to have a (tiled) Vector type at some point.